src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -499,16 +499,35 @@ protected function compileExtensions($value)
     /**
      * Compile Blade statements that start with "@".
      *
-     * @param  string  $value
+     * @param  string  $template
      * @return string
      */
-    protected function compileStatements($value)
-    {
-        return preg_replace_callback(
-            '/\B@(@?\w+(?:::\w+)?)([ \t]*)(\( ( (?>[^()]+) | (?3) )* \))?/x', function ($match) {
-                return $this->compileStatement($match);
-            }, $value
-        );
+    protected function compileStatements($template)
+    {
+        preg_match_all('/\B@(@?\w+(?:::\w+)?)([ \t]*)(\( ( [\S\s]*? ) \))?/x', $template, $matches);
+        for ($i = 0; isset($matches[0][$i]); $i++) {
+            $match = [
+                $matches[0][$i],
+                $matches[1][$i],
+                $matches[2][$i],
+                $matches[3][$i] ?: null,
+                $matches[4][$i] ?: null,
+            ];
+
+            // Here we check to see if we have properly found the closing parenthesis by
+            // regex pattern or not, and will recursively continue on to the next ")"
+            // then check again until the tokenizer confirms we found the right one
+            while (isset($match[4]) && Str::endsWith($match[0], ')') && ! $this->isProperMatch($match[0])) {
+                $rest = Str::before(Str::after($template, $match[0]), ')');
```

Comment:
I'm pretty confused by this Str::after statement. Can't the same statement appear twice in the same `$template`? Therefore, how would you know which one you're extracting after?

Reply:
I think the problem of having 2 exact same statements with different results is tested here: `testNestedForelseStatementsAreCompiled`, and that is why I used `Str::replaceFirst` and not `str_replace`.

Here `Str::after` gives us the rest of the template, starting at where the regex match has ended. (Str::after matches the first occurrence) then from that, we fetch the first closing parenthesis.

After we replace the statement, we go for the next one.